### ISSUE TITLE
Zero point fit

### ISSUE THREAD

**Issue by Yanshengyu0311** [Astro-Sean/autophot#8]
I'm Shengyu Yan, a Ph.D. student of Prof. Xiaofeng Wang, Tsinghua University. Yongzhi Cai suggest I use your pipeline to reduced the photometric image. Your python-based pipeline makes the data reduction much simple.

**Comment by Yanshengyu0311** [comment]
Hi Sean, Can you see my message?

**Comment by Astro-Sean** [comment]
Hi Shengyu - yes I can see your message - whats seems to be the issue?

**Comment by Yanshengyu0311** [comment]
I have some questions, I had posted some of my issues on the discussion board(https://github.com/Astro-Sean/autophot/discussions. ), you can check them first. Please give me some time to describe other issues.

**Comment by Astro-Sean** [comment]
Ah I see them now - my apologies for not noticing these earlier!

For the future it is best to report problems by opening an issue

**Comment by Yanshengyu0311** [comment]
Here is the zero point fit question:

Only one image has encountered this problem, But this photometric data point is important and valuable in the SN early raising phase. The zero point of the image is incorrect. Here is the zero-point fitting figure.

I guess it may be Due to the bad image quality and caused the catalog-image match error. It reports:

+---------------------------------+ |Matching catalog sources to image| +---------------------------------+  Removed 0 sources too close to boundary or off image Matching catalog to image: 181 / 182 :: Useful sources 128 / 182

 Median offset: 1.0 [ pixels ] / 1.7 [ arcsec ]
/home/ysy/anaconda3/envs/astroconda/lib/python3.7/site-packages/autophot/packages/psf.py:297: RuntimeWarning: All-NaN axis encountered 
pars.add('A',value = 0.75*np.nanmax(psf_image_bkg_free),
'float' object has no attribute 'shape'
Traceback (most recent call last):
  File "/home/ysy/anaconda3/envs/astroconda/lib/python3.7/site-packages/autophot/packages/psf.py", line 300, in build_r_table
  pars.add('x0',value = psf_image_bkg_free.shape[1]/2,
 AttributeError: 'float' object has no attribute 'shape'
** Fitting error - trying another source**
Matching catalog to image: 182 / 182 :: Useful sources 129 / 182 .. done 
Broken cutouts: 0
Not in correct location: 0
Not detected: 53
Saturated: 0
Error: 0

BTW, The fitting source for FWHM is 3-times less than other images. It reports:

Number of sources before cleaning [ 25.0 sigma ]: 60 
Removed 4 sources near boundary
Removed 13 crowded sources
Fitting source for FWHM: 43/43
Removed 34 saturated sources
WARNING: Input data contains invalid values (NaNs or infs), which were automatically clipped. [astropy.stats.sigma_clipping]
Input data contains invalid values (NaNs or infs), which were automatically clipped.
Removed 34 FWHM outliers
WARNING: Input data contains invalid values (NaNs or infs), which were automatically clipped. [astropy.stats.sigma_clipping]
Input data contains invalid values (NaNs or infs), which were automatically clipped.
Removed 35 median outliers
Useable sources found [ 25 sigma ]: 43
Removes 0 sources within minimum seperation [ 11 pixel ]

FWHM: 2.345 +/- 0.505 [ pixels ]

Seeing: 4.252 [ arcsec ]


And when building psf, it reports:

No PSF sources fitted
WARNING: Input data contains invalid values (NaNs or infs), which were automatically clipped. [astropy.stats.sigma_clipping]
Input data contains invalid values (NaNs or infs), which were automatically clipped.  
Mean r-band zeropoint: 20.219 +/- 0.981

I wonder if some parameters need to be modified in autophot_input. I had try to modify some parameters but failed. So I need your valuable advice!

**Comment by Yanshengyu0311** [comment]
> Ah I see them now - my apologies for not noticing these earlier!
> 
> For the future it is best to report problems by opening an issue

So kind of you. This is the first time I comment on Github. Some of the commentary etiquettes are still not very familiar.

**Comment by Astro-Sean** [comment]
Ah no problem - we're always learning - but I might recommend that in the future if you run into other problems open separate issues per problem - helps keep track of everything and may help others who run into the same problem 

So without seeing the image I can only make assumptions - the image quality is very bad ,seeing ~4 (typically a good image has ~1)

Also the FWHM is quite small - for a FWHM < ~3 you can run into sampling issues ([for some light reading](https://www.jstor.org/stable/10.1086/316460?seq=1#metadata_info_tab_contents))

So in this case the code detects a low fwhm and uses aperture photometry (which is what is done here) rather than build a PSF.

Also the error ``'float' object has no attribute 'shape' '' is related to a source that is trying to be measured but the cutout is all-nans, probably off image - this is fine.

Okay solutions to try:

1) There seems to be alot of saturated sources removed (34/40!) try setting:
autophot_input['source_detection']['remove_sat'] = False
to not remove saturated stars and see how things go 

2) Did you remove the cosmic rays?

3) Check the source_check output image and see what sources are being detection - does this seem resonable? 

4) if these don't work - would you be happy to send the image to me and I can debug things on my side and report back 
(I know there is an email issue so we'll work through this )

**Comment by Yanshengyu0311** [comment]
Thank you for taking the time to reply. Your suggestion seems to be of great value. I need some time to test out your suggestion.

Before running run_automatic_autophot, I did not remove the cosmic rays. And I haven't changed any parameters in autoiput about cosmic rays.I think the CRs have been removed automatically, and it reports: "Cosmic rays removed - image updated".

Here is the source_check_20211205_C_wcs.pdf
![图片](https://user-images.githubusercontent.com/54763954/155506249-fd9591b3-8bee-42b7-9f5d-a21b81cae783.png)
The the quality of source_check image looks not good. 

Thanks again!
